### PR TITLE
ci: eliminate 1.10 related logic, introduce 3.x

### PR DESCRIPTION
**Pull request by ylobankov**
- ci: not to push sources on `*-entrypoint` tag
- ci: eliminate 1.10 related logic for workflows 
- ci: introduce 3.x related logic for workflows

### CODE REVIEW

File: .github/workflows/source.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: source
 
 on:
   push:
-    tags:
-      - '**'
+    tags-ignore:
+      - '**-entrypoint'
```
Comment (Totktonada) [inline .github/workflows/source.yml:6]: Shouldn't we do the same for packages? I checked `centos_8.yml` and it doesn't contain the exception.

Comment (Totktonada) [reply .github/workflows/source.yml:6]: I see now, it is skipped in `.github/actions/pack-and-deploy/action.yml`.

Comment (ylobankov) [reply .github/workflows/source.yml:6]: We have such an exception here https://github.com/tarantool/tarantool/blob/master/.github/actions/pack-and-deploy/action.yml#L24, but your variant looks better. There is no need to run testing on an `*-entrypoint` tag.

Comment (Totktonada) [reply .github/workflows/source.yml:6]: Anyway, this is not a problem of this patchset, so I'll resolve the comment.

**Review by Totktonada — APPROVED** [review]
I see no conceptual problems.

I have not performed any testing or consistency check of the changes, my review is purely visual.

**Pull request merged**